pkgs/native_assets_builder/test_data/treeshaking_native_assets/hook/link.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+const packageName = 'treeshaking_native_assets';
+
+void main(List<String> arguments) async {
+  await link(arguments, (config, output) async {
+    final usedSymbols =
+        config.resources.map((resource) => resource.metadata.toString());
+    final dynamicLibrary = config.assets.firstWhere(
+        (asset) => asset.id.endsWith('src/${packageName}_bindings.dart'));
+    final staticLibrary =
+        config.assets.firstWhere((asset) => asset.id.endsWith('staticlib'));
+
+    final linkerScript = await _writeLinkerScript(usedSymbols);
+    await _treeshakeStaticLibrary(
+      usedSymbols,
+      linkerScript,
+      dynamicLibrary,
+      staticLibrary,
+    );
+    output.linkAsset(dynamicLibrary);
+    output.addDependency(config.packageRoot.resolve('hook/link.dart'));
+  });
+}
+
+Future<void> _treeshakeStaticLibrary(
+  Iterable<String> symbols,
+  Uri symbolsUri,
+  LinkableAsset dynamicLibrary,
+  LinkableAsset staticLibrary,
+) async {
+  File.fromUri(dynamicLibrary.file!).deleteSync();
```

Comment:
Let me try to understand why we need this.

If we rerun, but the previous link hook run has exactly the same hash so it's the same directory, the files are still there?

Should we consider clearing out the outdir in `native_assets_builder`? Don't we do that already?

Reply:
Wait, are we staying in the same directory for the build and link hook?
I believe `BuildConfigImpl.checksum(` should take in the `Hook` (build or link) when creating the checksum so that the directories are different for the hooks.
We can't rely on caching the build hook output if the link hook might destroy it.